src/core/scenario.jl

```diff
@@ -183,6 +184,13 @@ function Scenario(d::Dict; flex_hvac_from_json=false)
         storage_structs["ColdThermalStorage"] = ColdThermalStorage(params, financial, settings.time_steps_per_hour)
     end
     storage = Storage(storage_structs)
+    dc_coupled_pv = any([pv.dc_coupled_with_storage && pv.max_kw > 0 for pv in pvs])
+    if dc_coupled_pv && isempty(storage.types.dc_coupled)
+        throw(@error("To model PV that is DC coupled with storage (PV input dc_coupled_with_storage = true), you must also model a non-zero DC coupled storage system (set ElectricStorage input dc_coupled = true)"))
+    end
+    if !isempty(storage.types.dc_coupled) && !dc_coupled_pv
+        throw(@error("To model storage that is DC coupled with PV (ElectricStorage input dc_coupled = true), you must also model a non-zero DC coupled PV system (set PV input dc_coupled_with_storage = true)"))
+    end
```

Comment:
Does there need to be a check that if you're modeling multiple PVs, `dc_coupled_with_storage` is true for all of them?

Reply:
Nope, you can model a mixture!

Reply:
oh cool! How does this work if you're only modeling a single storage system? Or you would have to model multiple storage systems to do this?